### DISCUSSION TITLE
consent on page load with GTM

### DISCUSSION THREAD

**Discussion by majofi** [orestbida/cookieconsent#456] [General]
Hi,
I'm using cookieconsent together with an GTM setup and tried two ways to transfer the stored customer decision to the GTM consent settings:

a) in the onAccept function I do:
                    if (cc.allowedCategory('analytics')) {
                            gtag('consent', 'update', {
                                'analytics_storage': 'granted'
                            });
                        }

b) I have a script tag on the page that looks like this:
<script type="text/plain" data-category="analytics">
gtag('consent', 'update', {
    'analytics_storage': 'granted'
});
</script>

But both solution face the same problem:

1. They always happen after "Container Loaded" Event/Trigger, thus I can not fire any consent sensitive tag with that trigger, as the consent is not yet granted
2. using the "DOM Ready" Event/Trigger works most of the times, but also sometimes the consent updates happen a bit later and again tags are not fired.

The only solution I could think of at the moment is to manually read the cookie and extract the user settings during the "Content Initialization" event in GTM. Any other ideas / thoughts ?

**Comment by orestbida** [comment]
Hi @majofi,

you could try and load the gtm script when consent is expressed:

```html
<script 
    type="text/plain" 
    data-cookiecategory="necessary" 
    data-src="https://www.googletagmanager.com/gtag/js?id=TAG_ID"
></script>
```

Place this after all the other scripts with `data-cookiecategory`. That being said, this will very likely not work for categories enabled in a later moment.

Perhaps using the `dataLayer` object to send events to the template might be a better option.
